resources/mysql2_gem.rb

```diff
@@ -0,0 +1,25 @@
+provides :mysql2_gem
+resource_name :mysql2_gem
+
+property :compile_time, [true, false], default: false
+
+default_action :install
+
+action :install do
+  include_recipe 'mariadb::devel'
+  build_essential 'install_packages'
+  gem_package 'mysql2'
+end
+
+action :remove do
+  gem_package 'mysql2' do
+    action :remove
+  end
+end
+
+def after_created
+  return unless compile_time
+  Array(action).each do |action|
+    run_action(action)
+  end
+end
```

Comment:
Have given this some thought and wondering why this is a custom resource when it could be a recipe?

Also should we not bundle this and the devel recipe into the client recipe, so that if the client recipe is in the run list then the client bin and mysql2 gem is installed?